<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="6" passed="6" failed="0" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-10-13T11:37:20 IST" name="Default suite" finished-at="2024-10-13T11:37:20 IST" duration-ms="28">
    <groups>
    </groups>
    <test started-at="2024-10-13T11:37:20 IST" name="Default test" finished-at="2024-10-13T11:37:20 IST" duration-ms="28">
      <class name="NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest">
        <test-method is-config="true" signature="setUp()[pri:0, instance:NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest@17046283]" started-at="2024-10-13T11:37:20 IST" name="setUp" finished-at="2024-10-13T11:37:20 IST" duration-ms="5" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testAdd()[pri:0, instance:NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest@17046283]" started-at="2024-10-13T11:37:20 IST" name="testAdd" description="Test addition of two positive integers." finished-at="2024-10-13T11:37:20 IST" duration-ms="5" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAdd -->
        <test-method signature="testDivide()[pri:0, instance:NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest@17046283]" started-at="2024-10-13T11:37:20 IST" name="testDivide" description="Test division of two integers." finished-at="2024-10-13T11:37:20 IST" duration-ms="1" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testDivide -->
        <test-method signature="testDivideByZero()[pri:0, instance:NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest@17046283]" started-at="2024-10-13T11:37:20 IST" name="testDivideByZero" description="Test division by zero should throw IllegalArgumentException." finished-at="2024-10-13T11:37:20 IST" duration-ms="1" status="PASS">
          <exception class="java.lang.IllegalArgumentException">
            <message>
              <![CDATA[Division by zero is not allowed.]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.IllegalArgumentException: Division by zero is not allowed.
at NAGP_DevTestOps.NAGP_DevTestOps.Calculator.divide(Calculator.java:19)
at NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest.testDivideByZero(CalculatorTest.java:48)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.IllegalArgumentException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testDivideByZero -->
        <test-method signature="testMultiply()[pri:0, instance:NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest@17046283]" started-at="2024-10-13T11:37:20 IST" name="testMultiply" description="Test multiplication of two integers." finished-at="2024-10-13T11:37:20 IST" duration-ms="1" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testMultiply -->
        <test-method signature="testSubtractNegativeResult()[pri:0, instance:NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest@17046283]" started-at="2024-10-13T11:37:20 IST" name="testSubtractNegativeResult" description="Test subtraction resulting in a negative number." finished-at="2024-10-13T11:37:20 IST" duration-ms="1" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSubtractNegativeResult -->
        <test-method signature="testSubtractPositiveResult()[pri:0, instance:NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest@17046283]" started-at="2024-10-13T11:37:20 IST" name="testSubtractPositiveResult" description="Test subtraction resulting in a positive number." finished-at="2024-10-13T11:37:20 IST" duration-ms="0" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSubtractPositiveResult -->
      </class> <!-- NAGP_DevTestOps.NAGP_DevTestOps.CalculatorTest -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
